core/src/main/java/com/linecorp/armeria/server/AggregatingDecodedHttpRequest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import javax.annotation.Nonnull;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpObject;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.internal.common.stream.AggregatingStreamMessage;
+
+import io.netty.channel.EventLoop;
+
+final class AggregatingDecodedHttpRequest extends AggregatingStreamMessage<HttpObject>
+        implements DecodedHttpRequestWriter {
+
+    private final EventLoop eventLoop;
+    private final int id;
+    private final int streamId;
+    private final boolean keepAlive;
+    private final long maxRequestLength;
+    private final RequestHeaders headers;
+    private final RoutingContext routingCtx;
+    private final Routed<ServiceConfig> routed;
+    @Nullable
+    private ServiceRequestContext ctx;
+    @Nullable
+    private HttpHeaders trailers;
+    private long transferredBytes;
+
+    @Nullable
+    private HttpResponse response;
+    private boolean isResponseAborted;
+
+    AggregatingDecodedHttpRequest(EventLoop eventLoop, int id, int streamId, RequestHeaders headers,
```

Comment:
Since a request is aggregated without backpressure, I don't think we need to use `InboundTrafficController` for `AggregatingDecodedHttpRequest`. Did `integrate` mean to use a no-op `InboundTrafficController`?